xwiki-rendering-macros/xwiki-rendering-macro-message/src/main/java/org/xwiki/rendering/internal/macro/message/AbstractMessageMacro.java

```diff
@@ -73,4 +99,42 @@ public void prepare(MacroBlock macroBlock) throws MacroPreparationException
     {
         this.contentParser.prepareContentWiki(macroBlock);
     }
+
+    @Override
+    public List<Block> execute(BoxMacroParameters parameters, String content, MacroTransformationContext context) 
+        throws MacroExecutionException 
+    {
+        List<Block> boxFoundation = super.execute(parameters, content, context);
+        if (!boxFoundation.isEmpty() && this.iconName != null) {
+            Block defaultBox = boxFoundation.get(0);
+            // For an easier styling, we wrap the content and title together if they are non-empty and visible
+            if (defaultBox.getChildren().size() > 1) {
+                Block boxTextContent = new GroupBlock(defaultBox.getChildren());
+                defaultBox.setChildren(List.of(boxTextContent));
+            }
+            // Enhance the default box with an icon as the first element.
+            Block iconBlock = this.iconProvider.get(this.iconName);
+            // Provide an accessible name besides this icon
+            // This is the responsibility of the message macro and not the iconProvider which should only provide
+            // icons without any semantics
+            if (iconBlock.getClass() == ImageBlock.class) {
+                iconBlock.setAttribute("alt", this.iconPrettyName);
```

Comment:
This feels like something that should eventually be translated, but maybe you have another issue about that ?

Reply:
OK, I saw you overwritten all the macro on platform side. Not very consistent with the way icon were handled (since you overwrite everything, you could have done the same for icons as you did for the pretty name, not that I find it very clean of course...).

Reply:
> OK, I saw you overwritten all the macros on platform side. Not very consistent with the way icons were handled (since you overwrite everything, you could have done the same for icons as you did for the pretty name, not that I find it very clean of course...).

Yeah, from what I understood this is related to the discussion we had at https://github.com/xwiki/xwiki-platform/pull/3023#issuecomment-2155077952 . It's not the best, but for now it works and it doesn't lock us into a bad implementation.

Reply:
The problem with overwriting those macros is that eventually someone will add more stuff to those overwritten macros ("temporary" stuff tend to always stay longer than you initially, though...), and it will get harder and harder to get rid of them. An internal provider would really not cost much, and it does not lock us in any bad implementation either (but is much closer to the logic we eventually want).

Reply:
Okay, in the latest version, there is no overwriting of the macros at all.